epochStart/metachain/sovereignTrigger.go

```diff
@@ -112,6 +131,58 @@ func (st *sovereignTrigger) revert(header data.HeaderHandler) error {
 	return nil
 }
 
+// receivedBlock is a callback function when a new block is received
+// upon receiving checks if trigger can be updated to notify subscribed epoch change handlers
+func (st *sovereignTrigger) receivedBlock(headerHandler data.HeaderHandler, _ []byte) {
+	st.mutTrigger.Lock()
+	defer st.mutTrigger.Unlock()
+
+	header, ok := headerHandler.(data.MetaHeaderHandler)
+	if !ok {
+		return
+	}
+
+	if !header.IsStartOfEpochBlock() {
+		return
+	}
+
+	isMetaStartOfEpochForCurrentEpoch := header.GetEpoch() == st.epoch && header.IsStartOfEpochBlock()
+	if isMetaStartOfEpochForCurrentEpoch {
+		return
+	}
+
+	st.updateTrigger(header)
+}
+
+func (st *sovereignTrigger) updateTrigger(header data.MetaHeaderHandler) {
+	if !st.checkIfTriggerCanBeActivated(header) {
+		return
+	}
+
+	st.epochStartNotifier.NotifyEpochChangeConfirmed(header.GetEpoch())
+}
+
+func (st *sovereignTrigger) checkIfTriggerCanBeActivated(hdr data.HeaderHandler) bool {
+	missingMiniBlocksHashes, blockBody, err := st.peerMiniBlocksSyncer.SyncMiniBlocks(hdr)
```

Comment:
I don't think this is needed, as when processing the epoch start block -> the node will generate the peerminiblocks and can validate if the hash of the generated miniblock is the same as it is written in the header.